addons/l10n_co_pos/static/src/js/pos.js

```diff
@@ -2,31 +2,33 @@ odoo.define('l10n_co_pos.pos', function (require) {
 "use strict";
 
 var models = require('point_of_sale.models');
+const Registries = require('point_of_sale.Registries');
 
-models.PosModel = models.PosModel.extend({
-    is_colombian_country: function () {
+const L10nCoPosGlobalState = (PosGlobalState) => class L10nCoPosGlobalState extends PosGlobalState {
+    is_colombian_country() {
         return this.company.country.code === 'CO';
-    },
-});
+    }
+}
+Registries.PosModelRegistry.extend(models.PosGlobalState, L10nCoPosGlobalState);
```

Comment:
Why do we need to explicitly call models.PosGlobalState, and not just "PosGlobalState", which is the class extended.

Reply:
Can't we just require the PosGlobalState

Reply:
We always need the base class when extending.

Reply:
> Can't we just require the PosGlobalState
yes, we can.

Reply:
We can require `PosGlobalState` but it's not the extended version. It's still the base class. To get the extended version, we have to get it from the registry like so: `PosModelRegistry.get(PosGlobalState)`.

Do you want something such that when we do `const { PosGlobalState } = require('point_of_sale.models');`, `PosGlobalState` is already the extended version? I can't do that because when we need to have a way to get the base class so that we can extend it. However, I can do something like this:

```
const { PosGlobalState } = require('point_of_sale.models');

PosGlobalState.getBase() [or .Base] -> returns the base model which will be used for extending

PosGlobalState.getExtended() [or .Extended] -> returns the extended class which can be used for creating instances.
```

What do you think?

Reply:
Or do you mean I should do this? https://github.com/odoo/odoo/pull/82461/commits/fa2334540f12a687398dbf3e26946e3fe579e737